3rdparty/celt-0.11.0-build/celt-0.11.0-build.pro

```diff
@@ -64,13 +64,17 @@ unix {
 
 SOURCES *= bands.c celt.c cwrs.c entcode.c entdec.c entenc.c header.c kiss_fft.c laplace.c mathops.c mdct.c modes.c pitch.c plc.c quant_bands.c rate.c vq.c
 
-CONFIG(debug, debug|release) {
-  CONFIG += console
-  DESTDIR = ../../debug
-}
+CONFIG(custom_build_dir) {
+  OBJECTS_DIR = $$OBJECTS_DIR/3rdparty/celt-0.11.0
```

Comment:
Here, and everywhere else: you need to set not only OBJECTS_DIR, but all of OBJECTS_DIR, MOC_DIR, RCC_DIR and UI_DIR.

Maybe this should be moved to its own file, ala symbols.pri? We could name it destdir.pri or something. Then it could just use `OBJECTS_DIR = $$OBJECTS_DIR/$$TARGET`, if I am not mistaken. We don't have TARGETS with the same name, do we?

As for the current DESTDIR for non-CONFIG(custom_build_dir), I think we could just use an absolute path set in builddir.pri (included via compiler.pri).

Reply:
> As for the current DESTDIR for non-CONFIG(custom_build_dir), I think we could just use an absolute path set in builddir.pri (included via compiler.pri).

You could use `$$top_srcdir/release` or `$$top_srcdir/debug`... ... On Qt 5 😑

Darn.  So maybe there isn't a good way to do this in Qt 4 at all?

Reply:
> Here, and everywhere else: you need to set not only OBJECTS_DIR, but all of OBJECTS_DIR, MOC_DIR, RCC_DIR and UI_DIR.

It depends on the project requirements, in CELT's case only `OBJECTS_DIR` should be necessary.
But setting all of the paths anyway is a good idea. 

> Maybe this should be moved to its own file, ala symbols.pri? We could name it destdir.pri or something. Then it could just use OBJECTS_DIR = $$OBJECTS_DIR/$$TARGET, if I am not mistaken. We don't have TARGETS with the same name, do we?

Unfortunately both versions of CELT use **celt0** as `TARGET`, that's why I didn't do it.

>You could use $$top_srcdir/release or $$top_srcdir/debug... ... On Qt 5 :expressionless:
>
>Darn. So maybe there isn't a good way to do this in Qt 4 at all?

It seems like there isn't. The method described on Qt's website sounds more like a workaround than a solution.